pandas/io/parsers.py

```diff
@@ -234,8 +235,11 @@ def _read(filepath_or_buffer, kwds):
     if skipfooter is not None:
         kwds['skip_footer'] = skipfooter
 
-    filepath_or_buffer, _ = get_filepath_or_buffer(filepath_or_buffer,
-                                                   encoding)
+    compression = kwds.get('compression', None)
```

Comment:
put this directly in the call to ``get_filepath_or_buffer(....., compression=kws.get('compression', None)``